pysat/_meta.py

```diff
@@ -798,13 +801,37 @@ def drop(self, names):
 
         Parameters
         ----------
-        names : list-like
-            List of strings specifying the variable names to drop
+        names : str or list-like
+            String or list of strings specifying the variable names to drop
 
-        """
-
-        self.data = self._data.drop(names, axis=0)
+        Raises
+        ------
+        KeyError
+            If any of the keys provided in `names` is not found in the
+            lower dimensional data, higher dimensional data, labels, or
+            header data
 
+        """
+        # Ensure the input is list-like
+        names = listify(names)
+        for name in names:
+            if name in self._ho_data:
+                # Drop the higher dimension data
+                self._ho_data.pop(name)
+
```

Comment:
```suggestion
```

Reply:
This can only be applied after #1148 is merged in.